.github/workflows/build.yml

```diff
@@ -87,7 +87,7 @@ jobs:
           rust-cache: true
 
       - name: Cargo Build
-        run: cross build --target x86_64-unknown-linux-musl --workspace --release --all-targets --features openssl/vendored
```

Comment:
@rajatjindal FYI: I have removed these flags since they are unnecessary for building a Spin executable. I noticed this when the build was failing due to the `test-components` crate failing to build, but that crate does not need to be built to get a Spin executable.

Reply:
thank you for the heads up. these are also used in release workflow here https://github.com/fermyon/spin/blob/d41deb9eb33faeaa069d24a3922f26406c1cd2ca/.github/workflows/release.yml#L429C14-L429C20 

you may want to remove those as well in that case